client/pages/signup.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+
+export default class SignUp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: null,
+      passError: null
+    };
```

Comment:
Here you should have a default value for each input, every value you use from state needs to start with a default value. It will cause React not to re-render properly if you don't define all the state properties in the initial state.